posts/2021/07/moving-scipy-to-meson.md

```diff
@@ -0,0 +1,489 @@
+                "-O2",
+                "-g",
+                "-fvisibility-inlines-hidden",
+                "-std=c++17",
+                "-fmessage-length=0",
+                "-march=nocona",
+                "-mtune=haswell",
+                "-ftree-vectorize",
+                "-fPIC",
+                "-fstack-protector-strong",
+                "-fno-plt",
+                "-O2",
+                "-ffunction-sections",
+                "-pipe",
+                "-isystem",
+                "/home/rgommers/anaconda3/envs/scipy-meson/include",
+                "-DNDEBUG",
+                "-D_FORTIFY_SOURCE=2",
+                "-O2",
+                "-isystem",
+                "/home/rgommers/anaconda3/envs/scipy-meson/include",
+                "-fPIC",
+                "-DNPY_NO_DEPRECATED_API=NPY_1_9_API_VERSION"
+            ],
+            "sources": [
+                "/home/rgommers/code/bldscipy/scipy/sparse/sparsetools/bsr.cxx",
+                "/home/rgommers/code/bldscipy/scipy/sparse/sparsetools/csc.cxx",
+                "/home/rgommers/code/bldscipy/scipy/sparse/sparsetools/csr.cxx",
+                "/home/rgommers/code/bldscipy/scipy/sparse/sparsetools/other.cxx",
+                "/home/rgommers/code/bldscipy/scipy/sparse/sparsetools/sparsetools.cxx"
+            ],
+            "generated_sources": []
+        }
+    ],
+    "extra_files": [],
+    "subproject": null,
+    "installed": true,
+    "install_filename": [
+        "/usr/local/lib/python3.9/site-packages/scipy/sparse/_sparsetools.cpython-39-x86_64-linux-gnu.so"
+    ]
+}
+```
+This helps quickly pinpointing mistakes in your `meson.build` files. In
+addition, debugging potential issues in Meson itself, the generated
+`ninja.build` file is fairly readable too - its syntax is simple enough that
+for example missing dependencies can be found (e.g., `scipy.cluster`
+extensions depend on `scipy.linalg.cython_linalg`, and that dependency must
+be declared correctly).
+
+### Other benefits
+
+Other significant benefits include:
+
+1. Cross-compiling will become possible. For years we've told people *"sorry,
+   `distutils` wasn't really made for cross-compiling, let us know if you
+   have any luck".* As a result we've completely ignored users on some exotic
+   platforms (doing native ), and also spent a lot of time fighting with
+   different CI systems to do native builds. For example, we will likely want
+   to cross-compile to `aarch64` rather than struggle with underpowered ARM
+   hardware on Travis CI.
+2. Developers can use multiple builds at the same time. Because Meson builds
+   are out-of-tree by design, it is now easy to have for example a GCC build,
+   a Clang build, and a Python debug build in parallel in the same repo.
+3. More development tools work out of the box, e.g. `ccache` will just be
+   picked up if it's installed, no configuration needed. I also managed to
+   get ASAN to work with only a few small tweaks.
```

Comment:
Links to `ccache` and ASAN for newcomers might be nice

Reply:
thanks, added. and changed to AddressSanitizer to make clearer what it is without following the link